sloghandler.go

```diff
@@ -130,7 +130,7 @@ func (l *slogHandler) WithGroup(name string) slog.Handler {
 	copy := *l
 	if l.slogSink != nil {
 		copy.slogSink = l.slogSink.WithGroup(name)
-		copy.sink = l.slogSink
+		copy.sink = copy.slogSink
```

Comment:
Do we have a unit test for this?

Reply:
Clearly not. It was not clear to me that this field is ever used if slogSink is set - maybe we should set it nil?

Reply:
The other `With*` methods use `sink`, so `WithGroup("foo").WithName("bar")` probably was broken and dropped the group.

I asked about the unit test because I was wondering whether you had observed a failure or merely noticed it when looking at the code.

Reply:
I found it when single-stepping in the debugger. :)